### PR TITLE
Add upload button to my videos view, various Select fixes

### LINKED ISSUE #1270: Add upload button on MyVideosView, update tabs section
image
image
image
Designs: https://www.figma.com/file/LF7LOMhOA46bRcYHWlw1Bq/[Page]-My-videos?node-id=217%3A6627

Make sure that all the elements are in the right position.

### LINKED ISSUE #1401: TypeError: ref.scrollIntoView is not a function
When you click `Upload` button on unfilled form the following error appears in console
Zrzut ekranu 2021-09-23 o 14 25 42
:

### PR DESCRIPTION
**Pull request by drillprop**
Fix #1270 
Fix #1401

Also: 
- brought back forwardRef to Select component(saw some errors in the console earlier, focus on error didn't work correctly)
- some smaller input size and button size adjustments

### CODE REVIEW

File: src/shared/components/ButtonBase/ButtonBase.style.ts [hunk 1]
```diff
@@ -234,8 +234,8 @@ export const BorderWrapper = styled.div<Pick<ButtonBaseStyleProps, 'textOnly'>>`
   display: flex;
   align-items: center;
   box-sizing: border-box;
-  margin-top: -0.5px;
-  margin-bottom: -0.5px;
+  margin-top: ${({ textOnly }) => textOnly && '-0.5px'};
+  margin-bottom: ${({ textOnly }) => textOnly && '-0.5px'};
```
Comment (drillprop) [inline src/shared/components/ButtonBase/ButtonBase.style.ts:238]: just for clarification, the buttons had some weird heights. Large buttons had 47px, medium 39px and small 31. These lines should fix that, all the sizes should be divisible by 4 now.

File: src/shared/components/InputBase/InputBase.styles.ts [hunk 2]
```diff
@@ -28,9 +30,9 @@ export const FormGroup = styled.label<FormGroupProps>`
   textarea {
     transition: border ${transitions.timings.regular} ${transitions.easing};
     background: none;
-    padding: 10px ${sizes(4)};
+    padding: ${sizes(2)} ${sizes(4)};
     font-size: ${typography.sizes.body1};
-    line-height: 28px;
+    line-height: 30px;
```
Comment (rafalpawlow) [inline src/shared/components/InputBase/InputBase.styles.ts]:
```suggestion
    line-height: ${typography.lineHeights.body1}
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 3]
```diff
@@ -288,10 +310,12 @@ export const MyVideosView = () => {
         <>
           <TabsContainer>
             <Tabs initialIndex={0} tabs={mappedTabs} onSelectTab={handleSetCurrentTab} />
-            <SortContainer>
-              <Text variant="body2">Sort by</Text>
-              <Select value={sortVideosBy} items={SORT_OPTIONS} onChange={handleSorting} />
-            </SortContainer>
+            {mdMatch && sortVisibleAndUploadButtonVisible && sortSelectNode}
+            {smMatch && sortVisibleAndUploadButtonVisible && (
+              <Button to={absoluteRoutes.studio.editVideo()} icon={<SvgGlyphUpload />}>
```
Comment (rafalpawlow) [inline src/views/studio/MyVideosView/MyVideosView.tsx]: On designs there is different icon: `SvgGlyphAddVideo`

**Comment by rafalpawlow** [comment]
Something is wrong with button width
Zrzut ekranu 2021-09-23 o 14 54 06
Zrzut ekranu 2021-09-23 o 14 54 18

**Review by rafalpawlow — CHANGES_REQUESTED** [review]

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 4]
```diff
@@ -269,10 +272,29 @@ export const MyVideosView = () => {
...
   }
 
+  const sortVisibleAndUploadButtonVisible = isDraftTab ? !!drafts.length : !hasNoVideos
+
+  const sortSelectNode = (
+    <Select
+      size="small"
+      labelPosition="left"
+      label="Sort by"
+      helperText={null}
```
Comment (rafalpawlow) [inline src/views/studio/MyVideosView/MyVideosView.tsx]: This is not needed since we changed `helperText`
```suggestion
```

**Review by rafalpawlow — APPROVED** [review]

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 5]
```diff
@@ -269,10 +272,29 @@ export const MyVideosView = () => {
...
+      onChange={handleSorting}
+    />
+  )
+
   const mappedTabs = TABS.map((tab) => ({ name: tab, badgeNumber: tab === 'Drafts' ? unseenDrafts.length : 0 }))
   return (
     <StyledLimitedWidthContainer>
       <StyledText variant="h2">My videos</StyledText>
+      {!smMatch && sortVisibleAndUploadButtonVisible && (
+        <MobileButton to={absoluteRoutes.studio.editVideo()} icon={<SvgGlyphUpload />}>
```
Comment (rafalpawlow) [inline src/views/studio/MyVideosView/MyVideosView.tsx]: On designs there is different icon: `SvgGlyphAddVideo`

**Review by rafalpawlow — CHANGES_REQUESTED** [review]

**Comment by drillprop** [comment]
> Something is wrong with button width
> Zrzut ekranu 2021-09-23 o 14 54 06
> Zrzut ekranu 2021-09-23 o 14 54 18

I think it's a mistake in the design and I shouldn't change it. Normally medium-size button has 16px horizontal padding. In this view for some reason, it has only 12px. 
My videos: 
image
Web components: 
image

**Review by rafalpawlow — APPROVED** [review]

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 6]
```diff
@@ -288,10 +309,12 @@ export const MyVideosView = () => {
...
           <TabsContainer>
             <Tabs initialIndex={0} tabs={mappedTabs} onSelectTab={handleSetCurrentTab} />
-            <SortContainer>
-              <Text variant="body2">Sort by</Text>
-              <Select value={sortVideosBy} items={SORT_OPTIONS} onChange={handleSorting} />
-            </SortContainer>
+            {mdMatch && sortVisibleAndUploadButtonVisible && sortSelectNode}
+            {smMatch && sortVisibleAndUploadButtonVisible && (
+              <Button to={absoluteRoutes.studio.editVideo()} icon={<SvgGlyphAddVideo />}>
+                Upload Video
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
                Upload video
```

File: src/views/studio/MyVideosView/MyVideosView.tsx [hunk 7]
```diff
@@ -269,10 +272,28 @@ export const MyVideosView = () => {
...
+    />
+  )
+
   const mappedTabs = TABS.map((tab) => ({ name: tab, badgeNumber: tab === 'Drafts' ? unseenDrafts.length : 0 }))
   return (
     <StyledLimitedWidthContainer>
       <StyledText variant="h2">My videos</StyledText>
+      {!smMatch && sortVisibleAndUploadButtonVisible && (
+        <MobileButton size="large" to={absoluteRoutes.studio.editVideo()} icon={<SvgGlyphAddVideo />}>
+          Upload Video
```
Comment (kdembler) [inline src/views/studio/MyVideosView/MyVideosView.tsx]:
```suggestion
          Upload video
```

**Review by kdembler — CHANGES_REQUESTED** [review]
1. On 1024px the button wraps and creates bad visuals. Could we improve that?
Screenshot 2021-09-24 at 13 24 45

2. I've noticed that the tabs and `sort by` aren't aligned properly. That's because our implementation of Tabs differs slightly from the design. That is, we use a border which will make the tab label not centred inside its container. I guess you could change the border inside the tab to an inner box-shadow to achieve the same effect but without the misalignment?
Screenshot 2021-09-24 at 13 26 13

**Review by kdembler — APPROVED** [review]

**Pull request merged**